### PR TITLE
feat/TEN-132-authorization-cache-key-regex-masking

### PR DESCRIPTION
**Pull request by southbite**

### CODE REVIEW

File: packages/happn-3/lib/services/cache/cache-persist.js [hunk 1]
```diff
@@ -50,25 +50,31 @@ module.exports = class CachePersist extends require('./cache-static') {
...
+  set(key, data, opts = {}, transformedAlready, callback) {
     if (typeof opts === 'function') {
       callback = opts;
       opts = {};
     }
+    if (typeof transformedAlready === 'function') {
+      callback = transformedAlready;
+      transformedAlready = false;
+    }
+    opts = opts || {};
```
Comment (southbite) [inline packages/happn-3/lib/services/cache/cache-persist.js:62]: remove default of opts in line 53

File: packages/happner-2/lib/system/happn.js [hunk 2]
```diff
@@ -306,6 +308,27 @@ function __outboundLayer(message, callback) {
   callback(null, message);
 }
 
+function __initializeSecurityConfig(config) {
+  if (!config.happn.secure) {
+    return;
+  }
+  if (!config.happn.services.security) {
```
Comment (southbite) [inline packages/happner-2/lib/system/happn.js:315]: double-check that config.happn.services exists

File: packages/happner-2/lib/system/happn.js [hunk 3]
```diff
@@ -306,6 +308,27 @@ function __outboundLayer(message, callback) {
...
+  }
+  if (!config.happn.services.security.config) {
+    config.happn.services.security.config = {};
+  }
+  // this makes exchange response requests hittable by removing the response origin id ie: [0-9]+:set$/
+  config.happn.services.security.config.cache_checkpoint_authorization = {
+    keyTransformers: [
+      {
+        regex:
+          /^(?<keyMask>[a-zA-Z0-9-]+:\/_exchange\/responses\/[a-zA-Z0-9-_]+\/[a-zA-Z0-9-_]+\/[a-zA-Z0-9-_]+\/[a-zA-Z0-9-_]+)\/[0-9]+:set$/,
```
Comment (southbite) [inline packages/happner-2/lib/system/happn.js:326]: add a test in happner that checks the security cache contents

File: packages/happner-2/lib/system/happn.js [hunk 4]
```diff
@@ -306,6 +308,47 @@ function __outboundLayer(message, callback) {
...
+    return;
+  }
+  // no keyTransformer - by design (only really used for testing)
+  if (
+    config?.happn?.services?.cache?.config?.overrides?.checkpoint_cache_authorization
+      ?.keyTransformers === false
+  ) {
+    return;
+  }
+  if (!config.happn.services.cache) {
```
Comment (southbite) [inline packages/happner-2/lib/system/happn.js:322]: investigate lodash.set

**Review by leebow — APPROVED** [review]

**Pull request merged**